watersheds/_base.py

```diff
@@ -19,16 +19,14 @@ def construct_basin_dict():
 
     @staticmethod
     def find_basins_btw_source_mouth_in_basin_lv12(source_basin_id, mouth_basin_id, full_basin_dict, found_basins_id):
-        next_basin_id = full_basin_dict[source_basin_id]['NEXT_DOWN']
-        next_basin = full_basin_dict[next_basin_id]
-        # print(source_basin_id, 'next is', next_basin['HYBAS_ID'])
-        found_basins_id = found_basins_id + [next_basin['HYBAS_ID']]
-        if next_basin['HYBAS_ID'] == mouth_basin_id:
+        print('SOURCE, MOUTH:', source_basin_id, mouth_basin_id)
+        if source_basin_id == mouth_basin_id or source_basin_id == 0:
```

Comment:
what's "source_basin_id == 0" referring?

Reply:
`source_basin_id == 0` cover the case of `"NEXT_DOWN": 0`. This is saying that even if we don't end up at the mouth basin, we'll still have to stop if we reached the "last" basin (i.e. the ocean).

If we don't include this, we may continue even when we're at the "last" basin if we still didn't find our desired mouth basin. But that leads to an error since there's no basin with ID 0.

Reply:
But I am concerning about this case: we have both source_basin_id & mouth_basin_id, but the traversal stop by source_basin_id is 0, which suppose to end up at mouth_basin_id.

Reply:
Let's use this approach for now. But let's keep eyes on error cases for this.